fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -497,10 +507,25 @@ protected static void addCacheControlHeaders(final HttpServletResponse servletRe
             return;
         }
 
-        final FedoraResource mutableResource = resource instanceof NonRdfSourceDescription
-                ? ((NonRdfSourceDescription) resource).getDescribedResource() : resource;
-        final EntityTag etag = new EntityTag(mutableResource.getEtagValue());
-        final Date date = mutableResource.getLastModifiedDate();
+        final EntityTag etag;
+        final Date date;
+
+        // See note about this code in the javadoc above.
+        if (resource instanceof NonRdfSourceDescription) {
+            final NonRdfSourceDescription description = (NonRdfSourceDescription)resource;
+            // Use a weak ETag for the LDP-RS
+            etag = new EntityTag(description.getDescribedResource().getEtagValue(), true);
+            date = description.getDescribedResource().getLastModifiedDate();
+        } else if (resource instanceof FedoraBinary) {
```

Comment:
Yes the logic is invoked. Otherwise, when requesting a binary with an existing ETag:

    curl -i localhost:8080/fcrepo/rest/binary -H"If-None-Match: \"SOME-ETAG\""

you'd get a weak ETag in the response. My testing of this indicated that you receive a strong ETag. The same invocation on a container will result in a weak ETag.